File: James/230228_study/프로_138476_귤고르기.java
import java.util.*;

class Solution {
    static Map<Integer, Integer> map = new TreeMap<>();
    
    public int solution(int k, int[] tangerine) {
        int answer = 0;
        
        for(int key : tangerine) {
            map.put(key, map.getOrDefault(key, 0) + 1);
        }
        
        ArrayList<Integer> al = new ArrayList<>(map.keySet());
        Collections.sort(al, new userComparator());
        
        // System.out.println(al.toString());
        
        for(int key : al) {
            if(k <= 0)
                break;
            
            answer++;
            k -= map.get(key);
        }
        
        return answer;
    }
    
    public class userComparator implements Comparator<Integer> {
        @Override
        public int compare(Integer o1, Integer o2) {
            return map.get(o2).compareTo(map.get(o1)); // 귤의 크기의 개수로 정렬
        }
    } 
}

File: sammy/230302_study/프로_138476_귤 고르기.py
from collections import Counter

def solution(k, tangerine):
    answer = 0
    cnt=0 # 종류에 따라 귤의 개수 파악해서 더해나가기
    data=Counter(tangerine).most_common()
   
    while cnt<k:
        cnt+=data.pop(0)[1] # 가장 빈도수가 많은 값부터 pop하고 cnt에 더해나가기
        answer+=1 # 종류의 개수 +1

    return answer


# print(solution(6, [1, 3, 2, 5, 4, 5, 2, 3]), 3)
# print(solution(4, [1, 3, 2, 5, 4, 5, 2, 3]), 2)
# print(solution(2, [1, 1, 1, 1, 2, 2, 2, 3]), 1)
